src/api.js

```diff
@@ -0,0 +1,94 @@
+import { Promise } from 'es6-promise';
+
+class API {
+  constructor(addon) {
+    this.addon = addon;
+  }
+
+  // Currencies
+  getCurrencies(query) {
+    let self = this;
+
+    return new Promise((resolve, reject) => {
+      self.addon.request({
+        method: 'GET',
+        endpoint: 'currencies',
+        query: query,
+      }).then(response => resolve(response)).catch((err) => reject(err));
+    });
+  }
+
+  // Institutions
+  getInstitutions(query) {
```

Comment:
Maybe we should name the parameter here as "filters" ? to the caller these are filters being passed to the API?